grove-node-server-utils/filter.js

```diff
@@ -4,7 +4,7 @@ const queryBuilder = require('./ml-query-builder.service.js')({});
 var filter = (function() {
   var queryTextDefaultHandler = function(filter) {
     var q;
-    if (filter.constraint) {
+    if (filter.constraint !== undefined) {
```

Comment:
Perhaps we should also exclude `null`. What do you think @grtjn?

Reply:
If we do, we could include it everywhere

Reply:
Looks like the easiest way is to use `== null` and `!= null`, which covers null and undefined: https://stackoverflow.com/a/15992131

Reply:
== and != are against our lint styling. I agree on adding !== null. Best to be robust..